frontend/src/LineGraph.res

```diff
@@ -477,46 +525,72 @@ let make = React.memo((
     Js.Float.isNaN(x) ? "?" : Js.Float.toPrecisionWithPrecision(~digits, x)
   }
 
-  let lastValues =
-    dataSet->Belt.Array.map(x =>
-      x->BeltHelpers.Array.lastExn->DataRow.toValue->floatToStringHandleNaN
+  let handleLegendClick = (event: ReactEvent.Mouse.t) => {
+    let target = ReactEvent.Mouse.target(event)
+    let label = searchDataLabel(target)
+    let index =
+      labels
+      ->Belt.Option.getWithDefault([])
+      ->Belt.Array.getIndexBy(l => l == label)
+      ->Belt.Option.getWithDefault(-1)
+    setLinesState(_ =>
+      linesState->Belt.Array.mapWithIndex((idx, state) => idx == index ? !state : state)
     )
-  let isOverlayed = lastValues->Belt.Array.length > 1
+  }
+  let labels = labels->Belt.Option.getWithDefault([])
+  let lastValues = dataSet->Belt.Array.map(x => x->BeltHelpers.Array.lastExn->DataRow.toValue)
+  let lastValues = Belt.Array.zip(labels, lastValues)->Belt.Map.String.fromArray
+  let linesStateByLabel = Belt.Array.zip(labels, linesState)->Belt.Map.String.fromArray
+  let legendColorsByLabel = Belt.Array.zip(labels, legendColors)->Belt.Map.String.fromArray
```

Comment:
I'm not convinced that the `Belt.Map` are really helping here... There are a lot of noisy `Belt.Map...getWithDefault(label, ...)` below to get back the different values associated with a `label`. Would it be possible to structure the data as `(label * (lastValue * lineState * color)) array`? (then all the values would be accessible when we map below)

Reply:
Right, I wasn't very happy with how this code was written too, but hadn't really thought about improving. Thanks for your suggestion.